rx_signal_processing/rx_dsp_chain.cu

```diff
@@ -179,12 +179,34 @@ int main(int argc, char **argv){
       for (uint32_t beam_num=0; beam_num<rx_channel.beam_directions_size(); beam_num++) {
         // Go through each beam now and add the phases for each antenna to a vector.
         auto beam = rx_channel.beam_directions(beam_num);
+
+        // We are going to use two intermediate vectors here to rearrange the phase data so that
```

Comment:
I think you want these lines 183-187 to be above 179 for this to work correctly, and lines 203-209 outside of (after) the beam_num for loop. (to get: [_rx channel 1, beam 1, main data; rx channel 1, beam 2, main data; rx channel 1, beam 1, intf data; rx channel 1, beam 2, intf data; rx channel 2, beam 1, main data ... etc_]

Reply:
still this

Reply:
Agreed. I just pushed a fix to address this.